### PR TITLE
feat: Add support for tabular format

### PR DESCRIPTION
**Pull request by bahlo**
* Replace `MessageCode` w/ `str` as we have no way of knowing future message codes
* Adds support for the tabular format
* Adds iterator support to each table for better ergonomics

### CODE REVIEW

File: src/axiom_py/query/result.py [hunk 1]
```diff
@@ -151,17 +141,122 @@ class QueryLegacyResult:
...
+    name: str
+    # Sources contain the names of the datasets that contributed data to these
+    # results.
+    sources: List[Source]
+    # Order echoes the ordering clauses that was used to sort the results.
+    order: List[Order]
+    # Groups specifies which grouping operations has been performed on the
+    # results.
+    groups: List[Group]
+    range: Range
```
Comment (Licenser) [inline src/axiom_py/query/result.py:208]: range is optional
```suggestion
    range: Optional[Range]
```

File: src/axiom_py/query/result.py [hunk 2]
```diff
@@ -151,17 +141,122 @@ class QueryLegacyResult:
...
+    buckets: Optional[Bucket]
+    # Columns contain a series of arrays with the raw result data.
+    # The columns here line up with the fields in the Fields array.
+    columns: List[List[object]]
+
+    def events(self):
+        return ColumnIterator(self)
+
+
+class ColumnIterator:
```
Comment (Licenser) [inline src/axiom_py/query/result.py:221]: I love me some iterators <3

**Review by Licenser — COMMENTED** [review]
<3 tabularaza

File: src/axiom_py/query/result.py (hunk 1 above)
Comment (bahlo) [reply src/axiom_py/query/result.py]: Fixed in [cbb7584](https://github.com/axiomhq/axiom-py/pull/136/commits/cbb7584df0c782749daad2f68f5f28a71fe865f8) among other required/optional changes

File: src/axiom_py/query/result.py (hunk 2 above)
Comment (bahlo) [reply src/axiom_py/query/result.py:221]: Just wait for row-based support 😎

**Review by Licenser — APPROVED** [review]
<3

**Pull request merged**